streams/cluster/src/main/java/com/hortonworks/streamline/streams/cluster/service/EnvironmentService.java

```diff
@@ -434,6 +439,85 @@ public NamespaceServiceClusterMapping addOrUpdateServiceClusterMapping(
         return newMapping;
     }
 
+    public void injectProtocolAndPortToComponent(Map<String, String> configurations, Component component) {
```

Comment:
Copying my previous comment:
Not sure. Still not clear which is better: set vs add vs inject. Could you please describe why?